File: Labelizer-Frontend/src/app/_models/photo.ts
import { Dataset } from './dataset';
import { LabelBox } from './label-box';

export class Photo {
  id?: number;
  dataset?: Dataset;
  filename: string;
  extension: string;
  originalFilename: string;
  originalExtension: string;
  width: number;
  height: number;
  verified: boolean;
  labelBoxes?: LabelBox[];
  createdAt: Date;

  constructor(id: number, dataset: Dataset, filename: string, originalFilename: string) {
    this.id = id;
    this.dataset = dataset;
    this.filename = filename;
    this.originalFilename = originalFilename;
  }
}

File: Labelizer-Backend/src/main/java/com/raphaelcoutu/labelizer/entity/Photo.java
package com.raphaelcoutu.labelizer.entity;

import com.sun.org.apache.xpath.internal.operations.Bool;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@Table(name = "photo")
public class Photo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "dataset_id", nullable = false)
    Dataset dataset;

    @Column(name = "filename")
    private String filename;

    @Column(name = "extension")
    private String extension;

    @Column(name = "original_filename")
    private String originalFilename;

    @Column(name = "original_extension")
    private String originalExtension;

    @NotNull
    @Column(name = "width")
    private Integer width;

    @NotNull
    @Column(name = "height")
    private Integer height;

    @NotNull
    @Column(name = "verified", columnDefinition = "boolean default false")
    private Boolean verified;

    @OneToMany(mappedBy = "photo", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    private List<LabelBox> labelBoxes;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;
}
